gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -11,186 +11,232 @@ You may obtain a copy of the License at
-                // preferences, view
+        invokers: {
+            grantAuthorizationCode: {
+                funcName: "gpii.oauth2.authorizationService.grantAuthorizationCode",
+                args: ["{dataStore}", "{codeGenerator}", "{arguments}.0", "{arguments}.1", "{arguments}.2", "{arguments}.3"]
+                    // userId, clientId, redirectUri, selectedPreferences
+            },
+            addAuthorization: {
+                funcName: "gpii.oauth2.authorizationService.addAuthorization",
+                args: ["{dataStore}", "{codeGenerator}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // userId, clientId, selectedPreferences
+            },
+            userHasAuthorized: {
+                funcName: "gpii.oauth2.authorizationService.userHasAuthorized",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // userId, clientId, redirectUri
+            },
+            exchangeCodeForAccessToken: {
+                funcName: "gpii.oauth2.authorizationService.exchangeCodeForAccessToken",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // code, clientId, redirectUri
+            },
+            getSelectedPreferences: {
+                funcName: "gpii.oauth2.authorizationService.getSelectedPreferences",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1"]
+                    // userId, authDecisionId
+            },
+            setSelectedPreferences: {
+                funcName: "gpii.oauth2.authorizationService.setSelectedPreferences",
+                args: ["{dataStore}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                    // userId, authDecisionId, selectedPreferences
+            },
+            getUnauthorizedClientsForUser: {
+                funcName: "gpii.oauth2.authorizationService.getUnauthorizedClientsForUser",
+                args: ["{dataStore}", "{arguments}.0"]
+                    // userId
+            },
+            getAuthorizedClientsForUser: {
+                func: "{dataStore}.findAuthorizedClientsByUserId"
+                    // userId
+            },
+            revokeAuthorization: {
+                func: "{dataStore}.revokeAuthDecision"
+                    // userId, authDecisionId
+            },
+            getAccessTokenForOAuth2ClientIdAndGpiiToken: {
+                func: "{dataStore}.findAccessTokenByOAuth2ClientIdAndGpiiToken"
+                    // oauth2ClientId, gpiiToken
+            },
+            getAuthForAccessToken: {
+                func: "{dataStore}.findAuthByAccessToken"
+                    // accessToken
+            },
+            getClientByClientCredentialsAccessToken: {
+                func: "{dataStore}.findClientByClientCredentialsAccessToken"
+                    // accessToken
+            },
+            getClientCredentialsTokenPrivs: {
```

Comment:
There's no documentation for this as well as most of the other methods in this service. We need to produce at the very least a wiki page or .md file explaining what arguments the accept, and what they do